MaxText/utils/ckpt_conversion/to_maxtext.py

```
@@ -0,0 +1,184 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Converts a HuggingFace checkpoint to a MaxText compatible Orbax checkpoint."""
+
+import os
+import sys
+from typing import Sequence
+
+import numpy as np
+import jax
+import jax.numpy as jnp
+from absl import app
+from flax.training import train_state
+from transformers import AutoConfig, AutoModelForCausalLM
+
+from MaxText import checkpointing
+from MaxText import max_logging
+from MaxText import max_utils
+from MaxText import maxtext_utils
+from MaxText import pyconfig
+from MaxText import optimizers
+from MaxText.layers import models, quantizations
+from MaxText.train import save_checkpoint
+from MaxText.utils.ckpt_conversion.utils.param_mapping import HOOK_FNS, PARAM_MAPPING
+from MaxText.utils.ckpt_conversion.utils.utils import apply_hook_fns, HF_IDS
+
+
+def main(argv: Sequence[str]) -> None:
+  jax.config.update("jax_default_prng_impl", "unsafe_rbg")
+  os.environ["TF_CPP_MIN_LOG_LEVEL"] = "0"  # Suppress TensorFlow logging
+
+  hf_token = os.getenv("HF_AUTH_TOKEN")
+
+  config = pyconfig.initialize(argv)
+  model_id = HF_IDS[config.model_name]
+  max_utils.print_system_information()
+  if not config.base_output_directory:
+    output_directory = os.path.expanduser("~/.mt_output/")
+  else:
+    output_directory = config.base_output_directory
+
+  # Setup JAX distributed system and mesh
+  devices_array = maxtext_utils.create_device_mesh(config)
+  mesh = jax.sharding.Mesh(devices_array, config.mesh_axes)
+
+  # Load Hugging Face model, config, and state_dict
+  max_logging.log(f"Loading Hugging Face model: {model_id}...")
```

Do we have any mechanism to check the supported model_id? If a user provides an unsupported model, we could raise an error like: "Model {model_id} is not currently supported in MaxText. Supported models are: {list of supported models from ckpt_conversion}."

---

Added. Also revised the to_huggingface.py to make it consistent